boards: shields: add mchp_rnbd451_bt

JamesLiu-Microchip opened the pull request:
Add board support for the Microchip RNBD451 shield. This shield is compatible with a development board that includes a mikroBUS socket. This shield is intended to enable Bluetooth Low Energy functionality.

nandojve commented on boards/shields/mchp_rnbd451_bt/boards/sam_e54_xpro.overlay:
```diff
@@ -0,0 +1,43 @@
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+
```
Keep only 1 empty line here

nandojve commented on boards/shields/mchp_rnbd451_bt/boards/sam_e54_xpro.overlay:
```diff
@@ -0,0 +1,43 @@
+ * Copyright (c) 2025 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
```
This file should be dropped.
No board overlays inside shields.

JamesLiu-Microchip commented on boards/shields/mchp_rnbd451_bt/mchp_rnbd451_bt.overlay:
```diff
@@ -0,0 +1,18 @@
+};
+
+&sercom0{
+	bt_hci_uart: bt_hci_uart {
```
This shield is designed to extend Bluetooth functionality on vendor boards. It is currently supported and verified on specific devices from MCHP. After reviewing other Bluetooth-enabled shields, no one uses "type of shield" instead of an overlay. Could you provide a more concrete suggestion?
BTW, shield.yml is committed.

nandojve commented on boards/shields/mchp_rnbd451_bt/doc/index.rst:
```diff
@@ -0,0 +1,164 @@
+
+``cd zephyrproject``
+
+Append the above mentioned ``prj.conf`` configurations to each ``prj.conf`` in Example #1~#5.
```
@kartben ,

Is it already possible to define inside shield the correct `foo.conf` to automatically select the options when the shield be selected ?

kartben commented on boards/shields/mchp_rnbd451_bt/doc/index.rst:
```diff
@@ -0,0 +1,164 @@
+********
+
+The RNBD451 Add-on Board is an efficient low-cost development platform to evaluate and demonstrate
+the features, capabilities and interfaces of our Bluetooth® Low Energy module, RNBD451PE.
```
No "our" please. This is the Zephyr documentation

JamesLiu-Microchip committed:
dts: bindings: Add xprpo-header

The Xplained Pro layout provide a standard 20 pin header.  A board
can have one or more headers and can share pins.  The extension
headers are given names EXTn where n ϵ [1…7], n is determined by
which ID pin is connected to the embedded debugger.

Signed-off-by: James Liu <James.Liu2@microchip.com>

JamesLiu-Microchip committed:
boards: microchip: sam_e54_xpro: Add SERCOM ports

Expand SERCOM serial ports in the sam_e54_xpro board. This add and
enable sercom0, sercom1 and sercom5 ports which are located in the
xplained pro header. It makes use of the microchip,xpro-header to
easy use with shields.

Signed-off-by: James Liu <James.Liu2@microchip.com>

JamesLiu-Microchip committed:
boards: shields: Add mchp_rnbd451_bt

Add board support for the Microchip RNBD451 shield.
This shield is compatible with a development board that includes a
mikroBUS socket.

Signed-off-by: James Liu <James.Liu2@microchip.com>

JamesLiu-Microchip commented on boards/shields/mchp_rnbd451_bt/doc/index.rst:
```diff
@@ -0,0 +1,164 @@
+<https://docs.zephyrproject.org/latest/develop/getting_started/index.html>`_"
+in "`Developing with Zephyr
+<https://docs.zephyrproject.org/latest/develop/index.html>`_"
+for installation instructions.
```
related statements were removed.

nandojve commented on boards/microchip/sam/sam_e54_xpro/sam_e54_xpro.dts:
```diff
@@ -49,20 +33,92 @@
+			   <14 0 &portb 29 0>, /* SERCOM4 PAD[3] SPI MISO */
+			   <15 0 &portb 26 0>; /* SERCOM4 PAD[1] SPI SCK  */
+					       /* GND              */
+					       /* +3.3V            */
```
Needs fixes in all 3 headers
```diff
-					       /*           Shared */
+					       /*                  Shared */
-		gpio-map = <0  0 &portb  4 0>, /* ADC1/AIN[6]         */
+		gpio-map = <0  0 &portb  4 0>, /* ADC1/AIN[6]             */
...
			   <15 0 &portb 26 0>; /* SERCOM4 PAD[1] SPI SCK  */
-					       /* GND              */
-					       /* +3.3V            */
+					       /* GND                     */
+					       /* +3.3V                   */
```

JamesLiu-Microchip replied:
Thank you for your explanation, now I understand why and how to revise this.

nandojve reviewed (changes requested):
Hi @JamesLiu-Microchip ,

Thank you for the newer changes.

I found a few minor things, overall, LGTM!

Make sure you rebase before push.

@kartben ,

Is it possible to create vendor folder in shields ?
I see that it is starting to get a little messy for some vendors and I would like to avoid it.

JamesLiu-Microchip commented on boards/shields/mchp_rnbd451_bt/mchp_rnbd451_bt.overlay:
```diff
@@ -0,0 +1,5 @@
+ * Copyright (c) 2025 Microchip Technology Inc.
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
```
Thank you, nandojve, for your careful inspection, I have fixed the three places you mentioned.

nandojve commented:
> Hi @nandojve , I already use interactive rebase on my branch before force-pushing. Could you please explain in more detail what you mean by "rebase"? Did you mean I need to sync my fork with the latest main branch?

Hi @JamesLiu-Microchip ,

From my side I see some conflicts. You need to rebase with latest mainline, fix the conflicts, send another force push.
Merge commits are not allowed! Make sure you send only your commits on top of main branch : )

<img width="1222" height="1101" alt="image" src="https://github.com/user-attachments/assets/8e4647ff-23ec-4723-862c-bdbb044344c0" />

nandojve commented on boards/microchip/sam/sam_e54_xpro/sam_e54_xpro-pinctrl.dtsi:
```diff
@@ -7,13 +7,34 @@
 #include <dt-bindings/sam/sam_d5x_e5x/atsame54/atsame54p20a-pinctrl.h>
 
 &pinctrl {
+	sercom0_uart_default: sercom0_uart_default {
```
we need to follow (always)

`boards: <vendor>: <board_name>: Start-with-uppercase them-all-lower-case no-dot-at-end`

brief: (around 52 chars, max 72)
```diff
-boards: add sercom support based on xplained pro header
+boards: microchip: sam_e54_xpro: Add SERCOM ports
```

message: (max 72 - try use most you can in the line)
```diff
-sercom support based on xplained pro header. (around 52 chars, max 72)
+Expand SERCOM serial ports in the sam_e54_xpro board. This add and
+enable sercom0, sercom1 and sercom5 ports which are located in the
+xplained pro header. It makes use of the microchip,xpro-header to
+easy use with shields.
```

JamesLiu-Microchip committed:
boards: microchip: sam_e54_xpro: Binding to xprpo-header

Add xprpo-header support in sam_e54_xpro

Signed-off-by: James Liu <James.Liu2@microchip.com>

JamesLiu-Microchip commented on dts/bindings/gpio/microchip,xpro-header.yaml:
```diff
@@ -0,0 +1,48 @@
+# Copyright (c) 2025 Microchip Technology Inc.
```
I have modified the copyright in files new added. The new commit for special binding was added too.

nandojve reviewed (changes requested):
LGTM!

The PR need rebase to solve new conflicts.

@kartben , could you take another look?

JamesLiu-Microchip commented on boards/shields/mchp_rnbd451_bt/doc/index.rst:
```diff
@@ -0,0 +1,164 @@
+**********************
+To flash the built application, use the following command:
+
+``west flash``
```
Hi @kartben , Document was revised according to your review. could you please help to confirm again? Thank you.

nandojve commented:
ping @kartben

JarmouniA commented on boards/microchip/sam/sam_e54_xpro/sam_e54_xpro-pinctrl.dtsi:
```diff
@@ -7,13 +7,34 @@
 		};
 	};
 
+	sercom5_uart_default: sercom5_uart_default {
```
```suggestion
	sercom5_uart_default: sercom5-uart-default {
```

JamesLiu-Microchip replied:
Has committed the change. However, I only revised the node I added and preserved the original underscore style use by others.

JamesLiu-Microchip replied:
This change makes twister-build fail because the "sercom1_uart_default" node is already defined in two files in the tests/drivers/uart folder. The device tree compiler treats both as the same label and reports a duplicate definition. Do you have any suggestions?

Changing the definition in the tests/drivers/uart folder is not a good idea because it would alter definitions by other people and is not related to the original feature that was added.

kartben commented on boards/shields/mchp_rnbd451_bt/doc/index.rst:
```diff
@@ -0,0 +1,79 @@
+   :align: center
+   :alt: RNBD451 Add-on Board
+
+   **Figure 1: RNBD451 Add-on Board**
```
```suggestion
```

JamesLiu-Microchip replied:
Has committed the change.

nandojve commented:
ping @JarmouniA

kartben reviewed (commented):
a tiny typo left, sorry that I apparently missed it earlier

JamesLiu-Microchip commented on boards/shields/mchp_rnbd451_bt/doc/index.rst:
```diff
@@ -0,0 +1,77 @@
+.. zephyr-app-commands::
+   :app: samples/bluetooth/peripheral
+   :board: your_board_name
+   :shield: shield mchp_rnbd451_bt_mikrobus
```
Thanks for your check. Has committed the change.

kartben reviewed (approved):
@JamesLiu-Microchip thanks for your patience!
